api/src/main/java/org/apache/iceberg/SplittableScanTask.java

```diff
@@ -0,0 +1,60 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.apache.iceberg;
+
+public interface SplittableScanTask<ThisT> extends ScanTask {
+  /**
+   * Splits this scan task into several smaller scan tasks, each of {@code splitSize} size.
+   *
+   * @param splitSize the target size of each new scan task
+   * @return an Iterable of smaller tasks
+   */
+  Iterable<ThisT> split(long splitSize);
+
+  /**
+   * Checks if this task is adjacent to another.
+   *
+   * @param other another task
+   * @return whether this task is adjacent to another
+   */
+  boolean isAdjacent(ThisT other);
+
+  /**
+   * Combines two tasks into one.
```

Comment:
Is this only allowed for Adjacent Tasks? Or are we leaving that up to the implementation to decide?

Reply:
Hm, good question. I initially though about leaving that up to implementations but I am no longer sure. Maybe, we can change the doc to state that the passed task is adjacent. Then we won't have to check it again in classes that can only stitch adjacent tasks.

Reply:
Minor suggestion: Rename it to `combineAdjacentTask` if it is only for adjacent tasks?

Reply:
Renamed and changed the doc.